apps/builddao/widget/compose.jsx

```diff
@@ -1,9 +1,59 @@
 const [view, setView] = useState("editor");
-const [postContent, setPostContent] = useState("What do you have in mind?");
+const [postContent, setPostContent] = useState(props.template);
 const [hideAdvanced, setHideAdvanced] = useState(true);
 const [labels, setLabels] = useState([]);
 
-const name = Social.get(`${context.accountId}/profile/name`);
+function generateUID() {
+  const maxHex = 0xffffffff;
+  const randomNumber = Math.floor(Math.random() * maxHex);
+  return randomNumber.toString(16).padStart(8, '0');
+}
+
+function tagsFromLabels (labels) {
+  return labels.reduce((newLabels, label) => ({
+    ...newLabels,
+    [label]: "",
+  }), {})
+}
+
+const postToCustomFeed = ({ feed, text, labels }) => {
+  const postId = generateUID()
+  if (!labels) labels = []
+  return Social.set({
+    "update": {
+      [postId]: {
+        "": JSON.stringify({
+          type: "md",
+          text,
+          labels,
+        }),
+        "metadata": {
+          type: feed,
+          tags: tagsFromLabels(labels),
+        },
+      },
+    },
+    "post": {
+      "main": JSON.stringify({
+        type: "md",
+        text: `[EMBED](${context.accountId}/${feed}/${postId})`,
+      }),
+    },
+    "index": {
+      "every": JSON.stringify([
+        { key: feed, value: { type: "md" }},
+      ])
+    }
```

Comment:
Sorry, I wasn't clear, we should have this:

```
"index": {
      "post": JSON.stringify({
            { key: "main", value: { type: "md" }},
      }),
      "every": JSON.stringify(
        { key: feed, value: { type: "md" }}, (this should never be "main")
      )
    }
    ```
    
    Was only meant to replace the post/update index